cmp/filter/FilterChooserModel.js

```diff
@@ -0,0 +1,606 @@
+/*
+ * This file belongs to Hoist, an application development toolkit
+ * developed by Extremely Heavy Industries (www.xh.io | info@xh.io)
+ *
+ * Copyright © 2020 Extremely Heavy Industries Inc.
+ */
+
+import {FilterChooserFieldSpec} from '@xh/hoist/cmp/filter/impl/FilterChooserFieldSpec';
+import {HoistModel, managed, PersistenceProvider, XH} from '@xh/hoist/core';
+import {FieldFilter, FilterModel, parseFieldValue} from '@xh/hoist/data';
+import {action, observable} from '@xh/hoist/mobx';
+import {start, wait} from '@xh/hoist/promise';
+import {throwIf} from '@xh/hoist/utils/js';
+import {createObservableRef} from '@xh/hoist/utils/react';
+import {
+    compact,
+    differenceWith,
+    escapeRegExp,
+    flatten,
+    groupBy,
+    isEmpty,
+    isEqual,
+    isNaN,
+    isNil,
+    isPlainObject,
+    isString,
+    map,
+    partition,
+    sortBy,
+    take,
+    without
+} from 'lodash';
+
+@HoistModel
+export class FilterChooserModel {
+
+    @observable.ref value;
+    @observable.ref history;
+    @observable.ref options;
+
+    inputRef = createObservableRef();
+
+    /** @member {FilterModel} */
+    filterModel;
+    /** @member {Store} */
+    store;
+
+    /** @member {FilterChooserFieldSpec[]} */
+    @observable.ref fieldSpecs = [];
+
+    /** @member {string} */
+    valueSourceRecords;
+    /** @member {number} */
+    limit;
+    /** @member {number} */
+    maxHistoryLength;
+
+    /** @member {PersistenceProvider} */
+    @managed provider;
+    persistValue = false;
+    persistHistory = false;
+
+    /** @member {RawFilterChooserFieldSpec[]} */
+    _rawFieldSpecs;
+
+    // Option values with special handling
+    static TRUNCATED = 'TRUNCATED';
+    static SUGGEST_PREFIX = '*SUGGEST*';
+
+    /**
+     * @param c - FilterChooserModel configuration.
+     * @param {(FilterModel|Object)} c.filterModel - FilterModel, or config to create one.
+     * @param {Store} c.store - Store to use for Field resolution as well as extraction of available
+     *      Record values for field-specific suggestions. Note that configuring the store here does
+     *      NOT cause that store to be automatically filtered or otherwise bound to the FilterModel.
+     * @param {(string[]|FilterChooserFieldSpecConfig[])} [c.fieldSpecs] - specifies the Store
+     *      Fields this model will support for filtering and customizes how their available values
+     *      will be parsed/displayed. Provide simple Field names or `FilterChooserFieldSpecConfig`
+     *      objects to select and customize fields available for filtering. Optional - if not
+     *      provided, all Store Fields will be included with options defaulted based on their type.
+     * @param {string} [c.valueSourceRecords] - determines the set of Store Records used to extract
```

Comment:
This config was named `storeMode` in an earlier iteration, which I found to be a bit too general / non-descriptive. However I am not wild about my new name (it does imply that it would be an actual collection of records...) and am open to any suggestions for improvement.

I also am not personally clear on when you should override the default - we should provide guidance on when you would choose one over the other.